packages/rs-drive-abci/src/query/mod.rs

```diff
@@ -22,3 +22,1593 @@ impl<C> Platform<C> {
+            "/identities/by-public-key-hash",
+            "/proofs",
+        ];
+
+        paths.iter().for_each(|path| {
+                let result = platform.query(path, &vec![0; 8], &version);
+                assert!(result.is_ok());
+
+                let validation_result = result.unwrap();
+                let validation_error = validation_result.first_error().unwrap();
+
+                assert!(
+                    matches!(validation_error, QueryError::InvalidArgument(msg) if msg.contains("invalid query proto message"))
+                );
+            })
+    }
+
+    mod identity {
+        use crate::error::query::QueryError;
+        use crate::query::tests::assert_invalid_identifier;
+        use bs58::encode;
+        use dapi_grpc::platform::v0::{
+            get_identity_response, GetIdentityRequest, GetIdentityResponse,
+        };
+        use prost::Message;
+
+        const QUERY_PATH: &str = "/identity";
+
+        #[test]
+        fn test_invalid_identity_id() {
+            let (platform, version) = super::setup_platform();
+
+            let request = GetIdentityRequest {
+                id: vec![0; 8],
+                prove: false,
+            }
+            .encode_to_vec();
+
+            let result = platform.query(QUERY_PATH, &request, &version);
+            assert!(result.is_ok());
+            assert_invalid_identifier(result.unwrap());
+        }
+
+        #[test]
+        fn test_identity_not_found() {
+            let (platform, version) = super::setup_platform();
+
+            let id = vec![0; 32];
+            let request = GetIdentityRequest {
+                id: id.clone(),
+                prove: false,
+            }
+            .encode_to_vec();
+
+            let result = platform.query(QUERY_PATH, &request, &version);
+            assert!(result.is_ok());
+            let validation_result = result.unwrap();
+            let validation_error = validation_result.first_error().unwrap();
+
+            let error_message = format!("identity {} not found", encode(id).into_string());
+
+            assert!(matches!(
+                validation_error,
+                QueryError::NotFound(msg) if msg.contains(&error_message)
+            ));
+        }
+
+        #[test]
+        fn test_identity_absence_proof() {
+            let (platform, version) = super::setup_platform();
+
+            let id = vec![0; 32];
+            let request = GetIdentityRequest {
+                id: id.clone(),
+                prove: true,
+            }
+            .encode_to_vec();
+
+            let result = platform.query(QUERY_PATH, &request, &version);
+            assert!(result.is_ok());
+            let validation_result = result.unwrap();
+            let response =
+                GetIdentityResponse::decode(validation_result.data.unwrap().as_slice()).unwrap();
+
+            assert!(matches!(
+                response.result.unwrap(),
+                get_identity_response::Result::Proof(_)
```

Comment:
It's not a big deal but you want to match Ok result here as well instead of doing unwrap.